flake.nix

```diff
@@ -0,0 +1,56 @@
+{
+  description = "Build clojure projects with nix by creating a lockfile for maven and git dependencies";
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils, ... }: {
+
+    lib = rec {
+      customLocker =
+        { pkgs
+        , # The directory of this lockfile's project. If specified, a clean
+          # version of this repository (including uncommitted changes but
+          # without untracked files) will be available for lockfile generation
+          src ? null
+        , # The path to the lockfile, e.g. `"./deps.lock.json"`
+          lockfile
+        , # Specify the maven repositories to use, overriding the defaults
+          mavenRepos ?
+          [
+            "https://repo1.maven.org/maven2/"
+            "https://repo.clojars.org/"
+          ]
+        , # thecommand to produce the dependencies
+          command
+        }: {
+          locker = ((import ./default.nix { inherit pkgs; }).lockfile { inherit src lockfile mavenRepos; }).commandLocker command;
+          homeDirectory = ((import ./default.nix { inherit pkgs; }).lockfile { inherit src lockfile mavenRepos; }).homeDirectory;
+          shellEnv = ((import ./default.nix { inherit pkgs; }).lockfile { inherit src lockfile mavenRepos; }).shellEnv;
```

Comment:
can we DRY this up a little, for example by introducing a
```nix
let derivation = ((import ./default.nix { inherit pkgs; }).lockfile { inherit src lockfile mavenRepos; });
```
or something?

Reply:
Good call